src/case_gather/services.py

```diff
@@ -1,55 +1,149 @@
+                althingi_status=althingi_status,
+                althingi_link=case['althingi_link'],
+                status=getCaseStatus(althingi_status)
+            )
+
+            create_subjects(parliament_session, new_case, case['subject_names'])
+
+            createCaseCreators(case['case_creator_names'], new_case)
+
+
+def get_current_case_numbers_by_parliament_session(parliament_session):
+    """
+        Takes in a parliament_session, retrieves the cases connected to it
+        from the database and returns their ids
+    """
+    current_cases = Case.objects.filter(
+        parliament_session=parliament_session
+    )
 
     case_numbers = []
-    for case in cases_in_db:
+    for case in current_cases:
         case_numbers.append(case.number)
+    return case_numbers
 
-    try:
-        new_cases = xml_parser.get_case_data(session_number)
-    except Exception as e:
-        logger.error(e.message + '-' + traceback.format_exc())
 
-    #  Case has keys:
-    #  'number', 'name', 'case_type', 'case_status'
-    #  'rel_cases', 'subjects', 'session'
-    for case in new_cases:
-        logger.info(case)
-        if int(case['number']) in case_numbers:
-            logger.info('case is already in db')
+def getCaseStatus(althingi_status):
+    """
+        Takes in althingi_status (see: create_althingi_status_to_status_map())
+        and either returns the status we have defined to be equivalent
+        or returns Unknown as a default
+    """
+    althingi_status_to_status_map = create_althingi_status_to_status_map()
+
+    if althingi_status in althingi_status_to_status_map:
+        return althingi_status_to_status_map[althingi_status]
+    # Default status if the althingi status is not found in our database
+    return 'Unknown'
+
+
+def createCaseCreators(case_creator_names, case):
+    """
+    Takes in an array of strings which are names
+    Takes in a case, which is a Case object
+    It finds the ParliamentMember who has that name
+    and creates a CaseCreator object from the ParliamentMember
+    and the inputted Case.
+    If the ParliamentMember is not found, we log to Sentry as then
+    a new one has to be created in the database
+    """
+    for case_creator_name in case_creator_names:
+        parliament_member = ParliamentMember.objects.filter(name=case_creator_name)
+
+        if parliament_member.exists():
+            parliament_member = parliament_member.get()
+            CaseCreator.objects.create(
+                case=case,
+                parliament_member=parliament_member
+            )
+        else:
+            sentryLogger.logToSentry('Parliament member not found: ' + case_creator_name)
+
+def create_althingi_status_to_status_map():
+    """
+     Creates a map so that we can easily retrieve a more readable status from the
+     althingi_status. This is done as the althingi status is too complicated.
+     Key and Value are stored in database
+     Map will look something like:
+     {
```

Comment:
We need to be careful to check the date of this comment before trusting it too much.

You know, 6 months from now when we need to touch this code again